R/MPI.R

```diff
@@ -314,12 +314,32 @@ findPeaksPar <- function(arg) {
     if(params$lockMassFreq == TRUE){
         xRaw<-stitch(xRaw, AutoLockMass(xRaw))
     }
+    if(!is.null(unlist(params["polarity"])) & length(xRaw@polarity) > 1){
+    	## HPB: Check what this come as before production
+    	# xRaw<-stitch(xRaw, )
+    	xRaw.pol<- split(xRaw, f=xRaw@polarity)
+    	polarity<- match.arg(as.character(params["polarity"]), choices=levels(xRaw@polarity))
+    	ix.pol	<- grep(polarity, names(xRaw.pol))
+    	if(length(ix.pol) > 0){
+    		xRaw	<- xRaw.pol[[ix.pol]]
+    	} else{
+    		stop("Attempt to select polarity not present in file\n")
+    	}
+    }
     params["object"] <- xRaw
 
     ## remove parameters which are not used by method() from the parameter list
-    params["method"] <- params["id"] <- params["profmethod"] <- params["profparam"] <- params["includeMSn"] <- params["lockMassFreq"] <-  params["mslevel"] <- NULL
-
-    params["scanrange"] <- NULL ## avoid filtering scanrange twice, first in xRaw then in findPeaks
+    params["method"] 		<- NULL
+	arams["lockMassFreq"	<- NULL
```

Comment:
Hi, this line does not compile. I am currently fixing this, 
but I feel uncomfortable to push code that could not possibly have been run & tested. 
I will push to a different branch and will ask to test more.